drivers/regulator/regulator_npm1300.c

```diff
@@ -512,16 +519,57 @@ int regulator_npm1300_common_init(const struct device *dev)
 	return 0;
 }
 
+static int get_enabled_reg(const struct device *dev, uint8_t base, uint8_t offset, uint8_t mask,
+			   bool *enabled)
+{
+	const struct regulator_npm1300_config *config = dev->config;
+	uint8_t data;
+
+	int ret = mfd_npm1300_reg_read(config->mfd, base, offset, &data);
+
+	if (ret != 0) {
+		return ret;
+	}
+
+	*enabled = (data & mask) != 0U;
+
+	return 0;
+}
+
+static int get_enabled(const struct device *dev, bool *enabled)
+{
+	const struct regulator_npm1300_config *config = dev->config;
+
+	switch (config->source) {
+	case NPM1300_SOURCE_BUCK1:
+		return get_enabled_reg(dev, BUCK_BASE, BUCK_OFFSET_STATUS, BUCK1_ON_MASK, enabled);
+	case NPM1300_SOURCE_BUCK2:
+		return get_enabled_reg(dev, BUCK_BASE, BUCK_OFFSET_STATUS, BUCK2_ON_MASK, enabled);
+	case NPM1300_SOURCE_LDO1:
+		return get_enabled_reg(dev, LDSW_BASE, LDSW_OFFSET_STATUS, LDSW1_ON_MASK, enabled);
+	case NPM1300_SOURCE_LDO2:
+		return get_enabled_reg(dev, LDSW_BASE, LDSW_OFFSET_STATUS, LDSW2_ON_MASK, enabled);
+	default:
+		return -ENODEV;
```

Comment:
note: this is unreachable code in practice

Reply:
I guess it's needed to keep the compiler happy.

Reply:
yeah the way is used looks like an enum, ah well all good, let's go!